pkg/trigger/controller.go

```diff
@@ -133,14 +167,33 @@ func (mt *MigrationTrigger) enqueueResource(migration *migrationv1alpha1.Storage
 	mt.queue.Add(it)
 }
 
+func (mt *MigrationTrigger) addStorageVersion(obj interface{}) {
+	sv, ok := obj.(*apiserverinternalv1alpha1.StorageVersion)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("expected StorageVersion, got %#v", reflect.TypeOf(obj)))
+		return
+	}
+	mt.enqueueStorageVersion(sv)
+}
+
+func (mt *MigrationTrigger) updateStorageVersion(_ interface{}, obj interface{}) {
+	mt.addStorageVersion(obj)
+}
+
```

Comment:
I think we should handle deleteStorageVersion as well. The controller can garbage collect respective storageVersionMigration and storageState objects.